### PR TITLE
[COM-1678]: Add Autocomplete multiselect

### PR DESCRIPTION
**Pull request by Claratj**
**Checklist**

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

**Storybook Link** ([Link](https://5eed577398d08d0022b52d76-qxwlreplbp.chromatic.com/?path=/story/components-form-autocompletemultiselect--playground))

### CODE REVIEW

File: packages/components/src/theme/ComposerTheme/Components/Tag.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ const baseStyle = {
     color: 'text-secondary',
     border: '0.0625rem solid var(--chakra-colors-ui-element-outline-default)',
     borderRadius: '0.375rem',
+
```
Comment (guilean) [inline packages/components/src/theme/ComposerTheme/Components/Tag.ts:9]: 😿 @Claratj can we remove this empty line?

File: packages/components/src/theme/ComposerTheme/Components/Input.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ const iconContainerStyles = {
 export const iconStyles = {
   disabled: {
     color: 'text-disabled',
+    pointerEvents: 'none',
```
Comment (guilean) [inline packages/components/src/theme/ComposerTheme/Components/Input.ts:25]: Could you check if the pointerEvents is needed for the Input? I think we might only need it for the `Tag`

File: packages/components/src/components/form/AutocompleteMultiSelect/types.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import {
+  UseComboboxActions,
+  UseComboboxPropGetters,
+  UseComboboxProps,
+  UseMultipleSelectionActions,
+  UseMultipleSelectionPropGetters,
+  UseMultipleSelectionProps,
+} from 'downshift';
+import { FC, ReactElement, ReactNode } from 'react';
+// import { ResponsiveValue } from '@chakra-ui/react';
```
Comment (guilean) [inline packages/components/src/components/form/AutocompleteMultiSelect/types.ts:10]: Let's remove this line 🙌🏻

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+            cursor="pointer"
+            {...getToggleButtonProps()}
+          />
+        ) : (
+          <IconChevronDown
+            data-testid="cmpsr.autocompleteMultiSelect.chevron-down"
+            cursor="pointer"
+            {...getToggleButtonProps()}
+          />
+        )
```
Comment (guilean) [inline packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx:167]: Could we extract this into a constant component that will return the correct component based on the isOpen?

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  title: 'Components/Form/AutocompleteMultiSelect',
+  argTypes: {
+    size: {
+      options: autocompleteMultiSelectSizes,
+      control: { type: 'select' },
+    },
+  },
+} as Meta;
+
+const defaultItems = [
```
Comment (guilean) [inline packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.stories.tsx:18]: Could we try using an array of objects instead? I'd like to ensure we still have that functionality working 🙇🏻

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx (hunk 4 above)
Comment (Claratj) [reply packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx:167]: Is this what you had in mind? 🧐😊

File: packages/components/src/components/form/AutocompleteMultiSelect/types.ts (hunk 3 above)
Comment (Claratj) [reply packages/components/src/components/form/AutocompleteMultiSelect/types.ts:10]: done! 🚀

File: packages/components/src/theme/ComposerTheme/Components/Tag.ts (hunk 1 above)
Comment (Claratj) [reply packages/components/src/theme/ComposerTheme/Components/Tag.ts:9]: done!

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.stories.tsx (hunk 5 above)
Comment (Claratj) [reply packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.stories.tsx:18]: done!

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.test.tsx [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+        />
+      </AutocompleteMultiSelect>
+    );
+    const input = screen.getByPlaceholderText(/AutocompleteMultiSelect/);
+    fireEvent.focus(input);
+    const list = screen.getByRole('listbox');
+    expect(list.children).toHaveLength(1);
+    within(list).getByText(/No results custom/);
+  });
+  test('should call onInputValueChange when input value changes', () => {
```
Comment (guilean) [inline packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.test.tsx:92]: Rename test description and mock to be consistent with the property `onSelectedItemsChange`

Comment (Claratj) [reply packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.test.tsx:92]: done!

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+  const shouldShowClearButton = clearButtonConditions[clearButtonMode];
+  const inputSize = size ?? autocompleteMultiselectSize;
+  const isInputDisabled = isDisabled || autocompleteMultiSelectDisabled;
+
+  const onReset = useCallback(() => {
+    reset?.();
+    ref?.current?.focus?.();
+  }, []);
+
+  const pointerEventsStyle: {} = isInputDisabled ? { pointerEvents: 'none' } : {};
```
Comment (tkanzakic) [inline packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx:151]: nit: is the type definition as `{}` correct/needed?

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx
Comment (tkanzakic) [inline packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx:1]: nit: is it worthy to split this in multiple files?

**Review by tkanzakic — APPROVED** [review]

File: packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx (hunk 7 above)
Comment (Claratj) [reply packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx]: It was not really needed. Thanks!

Comment (Claratj) [reply packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx]: Updated to a more precise type. Thanks!

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**